### ISSUE TITLE
`uncapitalized-environment-variables` (`SIM112`) false positive on `https_proxy`, `http_proxy` and `no_proxy` variables

### ISSUE THREAD

**Issue by DetachHead** [astral-sh/ruff#8123]
```py
import os

os.environ["https_proxy"] # error
os.environ["http_proxy"] # error
os.environ["no_proxy"] # error
```
https://play.ruff.rs/75b318d8-1456-4743-a136-fe45150985ad

while there is no "official" standard, most applications use lowercase for these variables, so i think this rule should ignore them. or maybe a setting could be added for variable names to ignore

source:
- https://unix.stackexchange.com/a/212972
- https://about.gitlab.com/blog/2021/01/27/we-need-to-talk-no-proxy/#http_proxy-and-https_proxy

**Comment by charliermarsh** [comment]
This seems reasonable to me.
